File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
       int sum = 0, MAX = nums[0];
       for(int i: nums) {
           sum += i;
           MAX = max(MAX, sum);
           if (sum < 0) sum = 0;
       }
       return MAX;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int ans = -1000000, curr = 0;
        for(int i : nums) {
            curr += i;
            ans = Math.max(curr, ans);
            curr = Math.max(curr, 0);
        }
        return ans;
    }
}